poetry.lock

```diff
@@ -662,48 +686,49 @@ files = [
 
 [[package]]
 name = "mypy"
-version = "1.0.1"
+version = "1.7.1"
```

Comment:
I expect this is the cause of the CI errors!

Reply:
Updated PR to constrain the mypy more. Wasn't sure if you'd prefer an exact version constraint or not.

Reply:
I think it's ok to pin mypy on 1.0.1 because:
1. It's a dev dependency so it does not install through Hera on client side
2. Anecdotally any time I've updated mypy there was _some_ type error, somewhere. Pinning to 1.0.1 will help us decouple the mypy update from this CLI development and not make type changes in this PR